sonarts-core/src/rules/preferPromiseShorthandRule.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { SonarRuleVisitor } from "../utils/sonarAnalysis";
+import {
+  isIdentifier,
+  isFunctionLikeDeclaration,
+  isBlock,
+  isExpressionStatement,
+  isCallExpression,
+} from "../utils/nodes";
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "prefer-promise-shorthand",
+    description: "Shorthand promises should be used",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-4634",
+    type: "maintainability",
+    typescriptOnly: true,
+  };
+
+  public apply(sourceFile: ts.SourceFile): tslint.RuleFailure[] {
+    return new Visitor(this.getOptions().ruleName).visit(sourceFile).getIssues();
+  }
+}
+
+function message(value: ts.Node, action: PromiseAction) {
+  return `Replace this trivial promise with "Promise.${action}(${value.getText()})".`;
+}
+
+class Visitor extends SonarRuleVisitor {
+  protected visitNewExpression(node: ts.NewExpression) {
+    const executor = getPromiseExecutor(node);
+    if (executor) {
+      if (isFunctionLikeDeclaration(executor) && executor.body) {
+        const { parameters, body } = executor;
+        const [resolveParameterDeclaration, rejectParameterDeclaration] = parameters;
+        const resolveParameterName = getParameterName(resolveParameterDeclaration);
+        const rejectParameterName = getParameterName(rejectParameterDeclaration);
+
+        let bodyExpression = getOnlyBodyExpression(body);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Identifier 'bodyExpression' is never reassigned; use 'const' instead of 'let'. ([typescript:S3353](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3353))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=639&issues=AWNkE6wJ1_DDLGcJA7A1&open=AWNkE6wJ1_DDLGcJA7A1)

[ISSUE_KEY]: AWNkE6wJ1_DDLGcJA7A1